test/metabase/driver_test.clj

```diff
@@ -79,25 +80,63 @@
                          :field-definitions [{:field-name "foo", :base-type :type/Text}]
                          :rows              [["bar"]]}]}))
 
+(deftest can-connect-with-destroy-db-test
```

Comment:
Context behind this test: https://github.com/metabase/metabase/pull/35964#issuecomment-1824138064